src/libraries/System.Private.Xml/src/System/Xml/Core/XmlUtf8RawTextWriter.cs

```diff
@@ -710,6 +710,23 @@ private void FlushEncoder()
 
         }
 
+        internal override unsafe void WriteSpan(ReadOnlySpan<char> value)
+        {
+            Debug.Assert(value.Length != 0);
+            fixed (char* pSrc = value)
```

Comment:
```suggestion
            fixed (char* pSrc = &MemoryMarshal.GetReference(value))
```
As it's asserted that length > 0, we can use a "fast-path" here that doesn't have a redundant length check, as the `GetPinnableReference` has (which is generated by C# here for the fixed statement).